lib/src/facts/linux/filesystem_resolver.cc

```diff
@@ -152,7 +155,23 @@ namespace facter { namespace facts { namespace linux {
             }
 
             // Populate the size (the size is given in 512 byte blocks)
-            string blocks = file::read((path("/sys/class/block") / path(part.name).filename() / "/size").string());
+            // Check for /sys/class/block. Fall back to /sys/block if not present.
+            bs::error_code ec;
+            string blocks;
+            auto part_path = path(part.name).filename();
+            if (exists("/sys/class/block", ec) && !ec) {
+              blocks = file::read((path("/sys/class/block") / part_path / "/size").string());
```

Comment:
Indentation doesn't follow convention. Also, I'm mostly seeing that blkid doesn't find any devices (in containers, i.e. Travis and Docker).